sdk/storage/azure-storage-file/src/test/java/com/azure/storage/file/spock/FileAPITests.groovy

```diff
@@ -195,6 +196,23 @@ class FileAPITests extends APISpec {
         FileTestHelper.assertExceptionStatusCodeAndMessage(e, 416, StorageErrorCode.INVALID_RANGE)
     }
 
+    @Unroll
+    def "Upload data length mismatch"() {
+        given:
+        primaryFileClient.create(1024)
+        when:
+        primaryFileClient.uploadWithResponse(defaultData, size, 0, null)
+        then:
+        def e = thrown(UnexpectedLengthException)
+        e.getMessage().contains(errMsg)
+        cleanup:
+        defaultData.clear()
+        where:
+        size | errMsg
+        6 | "more than"
```

Comment:
These tests are throwing an `IllegalStateException` instead of `UnexpectedLengthException`

Reply:
Do you mean these tsts should throw `IllegalStateException ` ? I don't quite get why.

Reply:
Update the json files.